mango-python/bdgenomics/mango/QC.py

```diff
@@ -219,3 +221,136 @@ def plot(self, xScaleLog = False, yScaleLog = False, testMode = False, plotType=
+            mostSigEffect = most_sig_effect(effectList)
+            return mostSigEffect
+
+        z = variantRDD.toDF().rdd.map(lambda w: w.asDict()).map(lambda p: p['annotation']).map(lambda d: (d['attributes']['AF_NFE'], jp3(d)) )
+        self.b_data = z.filter(lambda f: f[1] == 'missense_variant').collect()
+        self.c_data = z.filter(lambda f: f[1] == 'synonymous_variant').collect()
+        self.sc = ss.sparkContext
+
+    def plot(self):
+        plt.rcdefaults()
+        plt.title("Allele Frequency Distribution by Functional Category")
+        fig, ax = plt.subplots(1,figsize=(14, 8))
+        b_data2= [ float(item[0]) for item in self.b_data if item[0] != "."]
+        c_data2= [ float(item[0]) for item in self.c_data if item[0] != "."]
+        ax.hist([b_data2,c_data2],bins=40, normed=1, label=['missense_variant','synonymous_variant'])
+        ax.legend(loc='upper right')
+        ax.set_yscale('log')
+        ax.yaxis.set_major_formatter(ticker.FuncFormatter(lambda y,pos: ('{{:.{:1d}f}}'.format(int(np.maximum(-np.log10(y),0)))).format(y)))
+        ax.yaxis.set_label_text("Percent Variants")
+        ax.xaxis.set_label_text("Allele Frequency")
+        plt.show()
+
+class VariantEffectCounts(object):
+
+    def __init__(self, ss, variantRDD, includeList = []):
+        """
+        Initializes a AlignmentDistribution class.
+        Computes the alignment distribution of multiple coverageRDDs.
+        :param SparkContext: the global SparkContext
+        :param alignmentRDDs: A list of bdgenomics.adam.rdd.AlignmentRDD objects
+        :param int bin_size: Division size per bin
+        """
+        if not includeList:
+            includeList = [
+                u'non_coding_transcript_exon_variant',
+                u'3_prime_UTR_variant',
+                u'5_prime_UTR_variant',
+                u'synonymous_variant',
+                u'stop_retained_variant',
+                u'conservative_inframe_insertion',
+                u'conservative_inframe_deletion',
+                u'5_prime_UTR_premature_start_codon_gain_variant',
+                u'disruptive_inframe_deletion',
+                u'disruptive_inframe_insertion',
+                u'missense_variant',
+                u'stop_lost',
+                u'start_lost',
+                u'splice_donor_variant',
+                u'splice_acceptor_variant',
+                u'splice_region_variant',
+                u'frameshift_variant',
+                u'stop_gained'
+            ]
+
+        effectCounts = variantRDD.toDF().select("annotation").rdd.map(lambda w: w[0].asDict()['transcriptEffects']) \
+            .flatMap(lambda m: m).map(lambda g: g.asDict()['effects']).flatMap(lambda a: a).filter(lambda i: i in includeList).countByValue()
+
+
+        self.sc = ss.sparkContext
+        self.effectCounts = effectCounts
+
+
+
+    def plot(self, xScaleLog = False, yScaleLog = False, testMode = False, plotType="I"):
+        """
```

Comment:
Can we add a normalization option for the second plot? (ie, print out fractions instead of counts?)